### ISSUE TITLE
Custom components should not check for no-access-key rules

### ISSUE THREAD

**Issue by pfdgithub** [jsx-eslint/eslint-plugin-jsx-a11y#849]
### Succeed
```jsx
<div />
<CustomComponent accessKey="h" />
```

### Fail
```jsx
<div accessKey="h" />
```

**Comment by ljharb** [comment]
I agree that this rule should only apply to HTML components; is that not the case?

**Comment by pfdgithub** [comment]
it doesn't seem that.
![image](https://user-images.githubusercontent.com/3262762/164954777-d346bbc5-7347-4100-b067-d115f6149e59.png)

**Comment by ljharb** [comment]
Seems like we should add an option to ignore custom components, and leave a TODO comment to enable it by default in the next major.

**Comment by pfdgithub** [comment]
Yes, all rules should not apply to custom components.

**Comment by V2dha** [comment]
@ljharb this option to ignore custom components, is it for `no-access-key` rule only?

**Comment by ljharb** [comment]
@V2dha This specific issue is only for that rule; but in general, most of the rules in this plugin should probably ignore custom components by default. Since doing so would be semver-major, they'd all need an option to allow for that.
